crates/apollo-compiler/test_data/diagnostics/0101_mismatched_variable_usage.txt

```diff
@@ -64,4 +64,15 @@ Error: variable `$intArg` of type `Int` cannot be used for argument `nonNullIntA
     │                                              ──────┬─────  
     │                                                    ╰─────── variable `$intArg` of type `Int` is declared here
 ────╯
+Error: expected value of type Boolean, found a variable
+    ╭─[0101_mismatched_variable_usage.graphql:51:50]
+    │
+ 51 │     nonNullBooleanListField(nonNullBooleanListArg: [$intArg])
+    │                                                     ───┬───  
+    │                                                        ╰───── provided value is a variable
+    │ 
+ 63 │     nonNullBooleanListField(nonNullBooleanListArg: [Boolean]!): Boolean
+    │                                                    ─────┬────  
+    │                                                         ╰────── expected type declared here as Boolean
+────╯
```

Comment:
I added this to show that the error message is very different depending on if you use the variable directly or nested inside a value... I think they should be treated the same